qemu/tests/usb_host.py

```diff
@@ -52,63 +65,57 @@ def usb_dev_unplug():
         usb_host_device_list = params["usb_host_device_list"].split(",")
         for dev in usb_host_device_list:
             vid, pid = dev.split(":")
-            monitor_add = "device_add usb-host,bus=usbtest.0,id=usbhostdev"
-            monitor_add += ",vendorid=%s" % vid
-            monitor_add += ",productid=%s" % pid
-            reply = vm.monitor.cmd(monitor_add)
-            negative_flag = False
-            for msg in usb_reply_msg_list:
-                if msg in reply:
-                    negative_flag = True
-                    break
-            if not negative_flag:
-                test.fail("Could not get expected warning"
-                          " msg in negative test, monitor"
-                          " returns: '%s'" % reply)
+            usb_params["vendorid"] = vid
+            usb_params["productid"] = pid
+            dev = qdevices.QDevice("usb-host", usb_params)
+            try:
+                vm.devices.simple_hotplug(dev, vm.monitor)
+            except QMPCmdError as detail:
+                logging.warn(detail)
+                for msg in usb_reply_msg_list:
+                    if msg in str(detail):
+                        break
+                else:
+                    test.fail("Could not get expected warning"
+                              " msg in negative test, monitor"
+                              " returns: '%s'" % detail)
```

Comment:
```
try:
    ***
except QMPCmdError as e:
    ***
else:
    test.fail("Hotplug *** should not success")
```

For the error info, better use `e.data["desc"]`.

Reply:
Updated, thanks!